docker-cloud/server_test.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/require"
+)
+
+var TestTable = map[string]struct {
+	endpoint string
+	expected string
+}{
+	"Index should return hello world": {
+		endpoint: "/",
+		expected: "Hello, world!",
+	},
+	"Ping should return pong": {
+		endpoint: "/ping",
+		expected: "Hello Pong!!",
+	},
+}
+
+func TestServer(t *testing.T) {
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err, "could not connect to Docker")
+
+	resource, err := pool.Run("docker-cloud", "latest", []string{})
+	require.NoError(t, err, "could not start container")
+
+	t.Cleanup(func() {
+		require.NoError(t, pool.Purge(resource), "failed to remove container")
+	})
+
+	for name, tt := range TestTable {
+		t.Run(name, func(t *testing.T) {
+
+			var resp *http.Response
+
+			err = pool.Retry(func() error {
```

Comment:
There's a limit of 60s by default, but it can be re-adjusted with `MaxWait` variable. Since this is a faily small container, I have set it to 20s now.